### PR TITLE
Add Kotlin I/O operations to fix #736

### LINKED ISSUE #736: [Kotlin] Add input/output
# Description

Input/Output for Kotlin aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Directions

* On the home page, on the right side (the reference sheet side), choose the language and version, and then the concept.
* The page says the data doesn't exist, but clicking the arrow for the template will provide the template and file for you to use.
* Save that file in `/web/thesauruses/[language]/[version]/io.json`.
* Fill out all of the `"code"` sections.
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then delete the `"code"` line.
* Add in `"comment":` if you wish to explain something in the code
* Delete the `"name":` fields before your PR
* Save, commit your code, push to your fork, then make a PR.
* Watch the checks to make sure they all pass. If they don't, click on "Details" and it should tell you what to modify.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by cduarte3**
## What GitHub issue does this PR apply to?

Resolves [#736 ](https://github.com/codethesaurus/codethesaur.us/issues/736#issue-2436514141)

## What changed and why?

As per my assigned issue, I've added Kotlin 1.5's input and output operations documentation.
- Added file io.json in the **web/thesauruses/Kotlin/1.5** directory
- Removed "name" fields in the file before committing and pushing PR as mentioned in the issue

## Checklist

   

- [X] I claimed any associated issue(s) and they are not someone else's
- [X] I have looked at documentation to ensure I made any revisions correctly
- [X] I tested my changes locally to ensure they work
- [X] For language files, I have validated the edited files are valid JSON and data shows up correctly
- [X] For website code edits, I have added or edited any appropriate unit tests for my changes

## Any additional comments or things to be aware of while reviewing?

[Kotlin docs](https://kotlinlang.org/docs/home.html) were used in resolving this issue.

Code was researched as best I could, if there are any typos, mistakes, adjustments to comments or syntax fixing needed, please let me know.

### CODE REVIEW

File: web/thesauruses/kotlin/1.5/io.json [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        ]
+    },
+    "read_char": {
+        "code": [
+        "val variable_name = readln()",
+        "val first_char = variable_name[0]"
+        ],
+        "comment": [
+        "First character can be read by accessing the first element of the string"
+        ]
```
Comment (geekygirlsarah) [inline web/thesauruses/kotlin/1.5/io.json]: Generally I've tried to keep the best practice that if you're writing code to replicate a function that doesn't exist, it shouldn't be in here. So if there's no built-in way to do that, it should probably be `not-implemented: true`

File: web/thesauruses/kotlin/1.5/io.json [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        "code": [
+        "java.io.File"
+        ]
+    },
+    "list_directory": {
+        "code": [
+        "val dir = File(\".\")",
+        "val files = dir.listFiles()",
+        "file?.forEach { println(it.name) }"
+        ]
```
Comment (geekygirlsarah) [inline web/thesauruses/kotlin/1.5/io.json]: Similarly, as a reference tool, this should probably be just `dir.listFiles` or like `File(".").listFiles()` to show the form of how it works, not the full example.

(Similar to others below)

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
Make tweak some of your examples and I'd be happy to merge this in! 

(If you're doing Hacktoberfest, it should still count towards Hacktoberfest since it was opened in October)

File: web/thesauruses/kotlin/1.5/io.json (hunk 1 above)
Comment (cduarte3) [reply web/thesauruses/kotlin/1.5/io.json]: Has been changed to:

image

File: web/thesauruses/kotlin/1.5/io.json (hunk 2 above)
Comment (cduarte3) [reply web/thesauruses/kotlin/1.5/io.json]: Has been changed to:

image

**Comment by cduarte3** [comment]
## Other changes:

Read_directory_info has been changed to reflect unavailability as there are no in built operations. Stats are not present, and size or length must be calculated manually.

image

<hr/>

Update_file has been changed to reflect unavailability as there are no set operations for this. Operations require modifying the lines of the file, and appending all plus newline char to the end of the list of strings.

New:

image

Old:

image

**Review by geekygirlsarah — APPROVED** [review]
Sweet! I'll merge this in if the checks pass. Thanks for contributing!

Happy (belated) Hacktoberfest!

**Comment by cduarte3** [comment]
Please let me know if everything looks okay now, I am happy to make any further changes.

The only section I am on the fence about now is the read_file_stream.

image

**Comment by geekygirlsarah** [comment]
The Docker check fails for a dependency issue, which isn't the fault here. So I'm merging it.

**Comment by cduarte3** [comment]
Alright, thanks for letting me know, and thanks for the help!

**Comment by geekygirlsarah** [comment]
> The only section I am on the fence about now is the read_file_stream.

Because of the two ways to do it? Yeah, this happens. I think this works!

**Pull request merged**